DropboxSync.BLL/Services/FileService.cs

```diff
@@ -0,0 +1,193 @@
+        /// Download the file from API and save it locally. Finally returns an <see cref="SavedFile"/> object containing needed data
+        /// </summary>
+        /// <param name="fileId">The ID of the file to download</param>
+        /// <returns><see cref="SavedFile"/> object if download and save was successfull. <c>null</c> Otherwise</returns>
+        public async Task<SavedFile?> DownloadFile(string fileId)
+        {
+            if (string.IsNullOrEmpty(fileId)) throw new ArgumentNullException(nameof(fileId));
+
+            SavedFile? finalOutput = null;
+
+            if (!await GetToken())
+            {
+                _logger.LogError("{date} | Could not retrieve Access Token to the API.", DateTime.Now);
+                return finalOutput;
+            }
+
+            string fileDownloadUrl = $"{API_BACKEND_URL}/api/resource/download?id={fileId}";
+
+            HttpResponseMessage response = await _httpClient.GetAsync(fileDownloadUrl);
+
+            if (response is null)
+            {
+                _logger.LogError("{date} | Something went wrong on api call to route \"{fileDownloadUrl}\"",
+                    DateTime.Now, fileDownloadUrl);
+                return finalOutput;
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.LogInformation("{date} | The API call was unsuccesfull. Reponse status is \"{responseCode}\"",
+                    DateTime.Now, response.StatusCode);
+
+                return finalOutput;
+            }
+
+            if (response.Content is null)
+            {
+                _logger.LogError("{date} | Response content is null!", DateTime.Now);
+                return finalOutput;
+            }
+
+
+            byte[] fileData = await response.Content.ReadAsByteArrayAsync();
+
+            if (fileData is null)
+            {
+                _logger.LogError("{date} | Byte array from file is null", DateTime.Now);
+                return finalOutput;
+            }
+
+            string? fileName = response.Content.Headers?.ContentDisposition?.FileName;
+
+            if (string.IsNullOrEmpty(fileName))
+            {
+                _logger.LogError("{date} | The file name for file with ID \"{fileId}\" could not be retrieved!",
+                    DateTime.Now, fileId);
+                return finalOutput;
+            }
+
+            fileName = fileName.Replace("\"", "");
+            string fileExtension = fileName.Split('.').Last();
+
+            // TODO : Make it really content type
+            string? contentType = response.Content.Headers?.ContentType?.MediaType;
+
+            if (string.IsNullOrEmpty(contentType))
+            {
+                _logger.LogError("{date} | The content type for file with ID \"{fileId}\" could not be retrieved!",
+                    DateTime.Now, fileId);
+                return finalOutput;
+            }
+
+            string filePath = $"{FILE_DOWNLOAD_DIR}\\{fileId}-{fileName}";
```

Comment:
There is another way that I discovered and used later. Seems that I forgot to change this like.

```csharp
string filePath = Path.Combine(FILE_DOWNLOAD_DIR, $"{fileId}-{fileName}");
```
or 

```csharp

string filePath = Path.Combine(FILE_DOWNLOAD_DIR, string.Join('-', fileId, fileName));

```

both functions will return the path in correct format in whichever Os is used

Reply:
Little Update. After our discussion, I decided to go on Path.Join() because of Microsoft's statement [here](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=net-6.0#remarks)